lms/djangoapps/commerce/api/v1/serializers.py

```diff
@@ -82,9 +87,27 @@ def validate(self, attrs):
 
         return attrs
 
-    def restore_object(self, attrs, instance=None):
-        if instance is None:
-            return Course(attrs['id'], attrs['modes'], attrs['verification_deadline'])
+    def create(self, validated_data):
+        """Create course modes for a course. """
+        course = Course(
+            validated_data["id"],
+            self._new_course_mode_models(validated_data["modes"]),
+            verification_deadline=validated_data["verification_deadline"]
+        )
+        course.save()
+        return course
+
+    def update(self, instance, validated_data):
+        """Update course modes for an existing course. """
+        validated_data["modes"] = self._new_course_mode_models(validated_data["modes"])
 
-        instance.update(attrs)
+        instance.update(validated_data)
+        instance.save()
         return instance
+
+    def _new_course_mode_models(self, modes_data):
```

Comment:
Why is this a method, rather than a standalone function? It doesn't use the `self` variable at all, so it shouldn't be a method.